src/WorkBC.Data/Migrations/20240413000723_InsertNocCodes2021.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Migrations;
+using System.Text.Json;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System;
+using WorkBC.Data.Model.JobBoard;
+using Microsoft.EntityFrameworkCore.Metadata.Internal;
+using WorkBC.Data.Model.Enterprise;
+using System.IO;
+using Microsoft.VisualBasic;
+using System.Text;
+using System.Linq;
+
+#nullable disable
+
+namespace WorkBC.Data.Migrations
+{
+    public partial class InsertNocCodes2021 : Migration
+    {
+        private readonly HttpClient _httpClient;
+        private readonly string _ssotUrl;
+
+        public InsertNocCodes2021() : base()
+        {
+            _httpClient = new HttpClient();
+            _ssotUrl = Environment.GetEnvironmentVariable("SSOT_URL");
```

Comment:
I thought we were adding an entry to the `appsettings.json` file.

Reply:
Also, notice that the config variable in `appsettings.json` will be the base URL of the SSOT API: e.g. `http://localhost:3000`. You need to append the correct endpoint here, which is `/nocs_nocs`.

Reply:
Modified the ReadMe file with instruction to set the environment variable for SSOT URL. It is a one-time activity before running any of the migrations related to 2021 NOC Codes. The command has to be run on Package manager console.

Reply:
I added my comments to the README commit. I still don't think the approach is the best one.

Reply:
Based on our conversation right now, I understand that the `appsettings.json` configuration file is not available to the migrations. [I confirmed this on Stack Overflow](https://stackoverflow.com/questions/41020640/how-to-get-settings-from-appsettings-json-in-a-referenced-net-framework-project).

In that case, let's proceed with `SSOT_URL` but point to the base URL of the SSOT API. Each migration can add its own endpoint to get the full URL.

Reply:
As discussed, have updated the code to derive the data from the new SSOT URL by setting it in Environment variable.